### ISSUE TITLE
[java] UnsynchronizedStaticFormatter cross-class violations not detected

### ISSUE THREAD

**Issue by coladict** [pmd/pmd#3163]
**Affects PMD Version:**  6.29.0

**Rule:** UnsynchronizedStaticFormatter

https://pmd.github.io/latest/pmd_rules_java_multithreading.html#unsynchronizedstaticformatter

**Description:**

False negatives. Does not detect when static formatter is used outside of its declared class.

**Code Sample demonstrating the issue:**

```java
// in Consts.java
import java.text.SimpleDateFormat;

public class Const {
	static final SimpleDateFormat SDF = new SimpleDateFormat("yyyy-MM-dd");
}
```
```java
// in Main.java
import java.util.Date;

public class Main {
	public void doStuff() throws Exception {
		Date date = Const.SDF.parse("2020-01-01");
		Const.SDF.format(date);
	}
}

```

**Expected outcome:**

PMD should detect two violations when invoking parse and format.

It's a false-negative.

**Running PMD through:** *Maven*

**Comment by jsotuyod** [comment]
Just to be clear, this violation should be reported directly on the `SDF` declaration, and not on the usages; and should be reported even if `Main` was synchronizing around it's usage.

Since the field is not private, it's impossible to guarantee all usages (current and future) are using the same lock even if they are trying. The only safe way to have this is to either remove the field and let everyone use their own, or expose thread-safe utility methods rather than the raw formatter instance.
